pkg/kubelet/dockershim/network/hostport/hostport_manager.go

```diff
@@ -152,10 +152,17 @@ func (hm *hostportManager) Add(id string, podPortMapping *PodPortMapping, natInt
 
 		// DNAT to the podIP:containerPort
 		hostPortBinding := net.JoinHostPort(podIP, strconv.Itoa(int(pm.ContainerPort)))
-		writeLine(natRules, "-A", string(chain),
-			"-m", "comment", "--comment", fmt.Sprintf(`"%s hostport %d"`, podFullName, pm.HostPort),
-			"-m", protocol, "-p", protocol,
-			"-j", "DNAT", fmt.Sprintf("--to-destination=%s", hostPortBinding))
+		if pm.HostIP == "" || pm.HostIP == "0.0.0.0" || pm.HostIP == "::" {
```

Comment:
Does kubelet know to use iptables for IPv6 if HostIP is `::` here?

Reply:
kubelet is ALWAYS single stack as today, and its IP family depends on the nodeIP, 

https://github.com/kubernetes/kubernetes/blob/c236285708d43ad7535a37cbff41fa380afc199d/pkg/kubelet/kubelet.go#L468-L473

The hostport_manager has its own iptables handler, but depends on the kubelet for some rules, however, due to this https://github.com/kubernetes/kubernetes/issues/82125 , those rules are added by kube-proxy

Now that you mention it, without kube-proxy this won't work for dual-stack @danwinship